### DISCUSSION TITLE
type error in doc

### DISCUSSION THREAD

**Discussion by DevEatSleep** [geopandas/geopandas#3021] [General]
In the file https://geopandas.org/en/stable/docs/user_guide/data_structures.html , i think that the sentence "Note that all entries in a [GeoSeries](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.html#geopandas.GeoSeries) need not be of the same geometric type, although certain export operations will fail if this is not the case." is wrong.

It is not "need not be" (which means nothing) but "need to be".

**Comment by gepcel** [comment]
No, I don't think it's a type of error. It means exactly what it means. eg:

```python
from shapely.geometry import Point, Polygon
p1 = Point(1, 2)
p2 = Polygon(((1,2),(2,2),(2,1),(1,2)))
ds = gpd.GeoSeries([p1, p2])
df.plot()
plt.show()
df.to_file('output.shp')
```

A GeoSerie contains a point and a polygon can be created successfully, can be plotted successfully, but can't be exported to a shapefile. Can be exported to some other formats, but it depends.

**Comment by martinfleis** [comment]
@gepcel is right that we do allow multiple geometry types but @DevEatSleep is right that the sentence is not really English. It should be "Note that all entries in a [GeoSeries](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.html#geopandas.GeoSeries) do not need to be of the same geometric type..."

If anyone wants to open a PR with a fix, I'll be happy to merge it.

**Reply by DevEatSleep** [reply]
thanks @martinfleis , the sentence is clearer now.

**Comment by gepcel** [comment]
> is not really English.

Haha, I don't know English well enough to tell the difference between "need not to be" and "do not need to be". Sorry.

**Comment by amichuda** [comment]
Just wanted to add my two cents, but don't have an opinion on whether this
gets changed either way.

"need not be" is a bit of an old or maybe fancy way of saying "does not
need to be," but it is in fact a correct way of saying it.

**Comment by martinfleis** [comment]
Fixed in #3022
